ast_comments.py

```diff
@@ -111,18 +113,69 @@ def _get_tree_intervals(
             if items := getattr(node, attr, None):
                 if not isinstance(items, Iterable):
                     continue
-                attr_intervals.append((*_get_interval(items), attr))
+                attr_intervals.append(_post_extend_interval((*_get_interval(items), attr), source))
         if attr_intervals:
-            low = node.lineno if hasattr(node, "lineno") else min(attr_intervals)[0]
-            high = (
-                node.end_lineno
-                if hasattr(node, "end_lineno")
-                else max(attr_intervals)[1]
-            )
+            # If the parent node hast lineno and end_lineno we extend them too, because there could be comments at the
+            #  end not covered by the intervals gathered in the attributes
+            if hasattr(node, "lineno") and hasattr(node, "end_lineno"):
+                low, high, _ = _post_extend_interval((node.lineno, node.end_lineno, ''), source)
+                node.lineno = low
+                node.end_lineno = high
+                # also update the end col offset corresponding to the new line
+                node.end_col_offset = len(source.split('\n')[high - 1])
+            else:
+                low = min(node.lineno, min(attr_intervals)[0]) if hasattr(node, "lineno") else min(attr_intervals)[0]
+                high = max(node.end_lineno, max(attr_intervals)[1]) if hasattr(node, "end_lineno") else max(attr_intervals)[1]
+
             res[(low, high)] = {"intervals": attr_intervals, "node": node}
     return res
 
 
+### Try to move lower bound lower and upper bound higher while not going out of bounds concerning the current block
+def _post_extend_interval(
+        interval: Tuple[int, int, str],
+        code: str
+) -> Union[List[Tuple[int, int]], ast.AST]:
+    lines = code.split('\n')
+    lines.insert(0, '')
```

Comment:
please, add a comment about this "insert"

Reply:
Added explanation this line is neccessary
 - it is easier to add a line than to use everywhere lineno - 1